src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -607,34 +607,82 @@ internal void VerifyWrite(int originalDepth, Utf8JsonWriter writer)
 #nullable restore
             JsonSerializerOptions options);
 
-        internal virtual T ReadAsPropertyName(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        /// <summary>
+        /// Reads a value from a JSON property name.
+        /// </summary>
+        /// <param name="reader">The <see cref="Utf8JsonReader"/> to read from.</param>
+        /// <param name="typeToConvert">The <see cref="Type"/> being converted.</param>
+        /// <param name="options">The <see cref="JsonSerializerOptions"/> being used.</param>
+        /// <returns>The value that was converted.</returns>
+        public virtual T ReadAsPropertyName(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
I probably just need more coffee, but.. what does it mean to "read a value from a property name"?  I've not looked at the implementation, but just looking at the method signature, I don't understand what the purpose of this method is.

Reply:
For context: https://github.com/dotnet/runtime/issues/46520#issuecomment-896196155. The method is meant to convert a JSON property name into a dictionary key.

Reply:
If it's clear to everyone else, then ignore me, but how does anything about this name, signature, or description in comments imply that? :smile:

Reply:
The previous suggestion was "ReadDictionaryKey" or equivalent. We really mean "ReadJsonPropertyThatIsGoingToBeUsedAsADictionaryKey()".

Reply:
We also considered using the following API shape
```csharp
public string WriteToJsonString(T value, JsonSerializerOptions options);
public T ReadFromJsonString(string jsonString, JsonSerializerOptions options);
```
however materializing the intermediate string would be suboptimal.